### PR TITLE
Document PEP 698 and other new typing features in What's New

### PR DESCRIPTION
**Pull request by JelleZijlstra**
cc @stroxler 

----
:books: Documentation preview :books:: https://cpython-previews--104957.org.readthedocs.build/

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
I looked over `git diff 3.11 3.12 Lib/typing.py` and noticed two more changes that deserve to be mentioned: @adriangb's work on `__orig_bases__` and @debonte's addition of `frozen_default`.

File: Doc/whatsnew/3.12.rst [hunk 1]
```diff
@@ -270,7 +270,35 @@ typed dictionaries::
...
-(PEP written by Franek Magiera)
+(Contributed by Franek Magiera in :gh:`103629`.)
+
+PEP 698: Override Decorator for Static Typing
+---------------------------------------------
+
+A new decorator :func:`typing.override` has been added to the :mod:`typing`
+module. It indicates to type checkers that the method is intended to override
+a method in a superclass. This allows type checkers to catch mistakes where
+a method that is intended to override a base class does not in fact do so.
```
Comment (AlexWaygood) [inline Doc/whatsnew/3.12.rst]:
```suggestion
a method that is intended to override something in a base class
does not in fact do so.
```

File: Doc/whatsnew/3.12.rst [hunk 2]
```diff
@@ -755,6 +778,13 @@ typing
   or more members may be slower than in Python 3.11. (Contributed by Alex
   Waygood in :gh:`74690` and :gh:`103193`.)
 
+* All :data:`typing.TypedDict` and :data:`typing.NamedTuple` classes now have the
+  ``__orig_bases__`` attribute. (Contributed by Adrian Garcia Badaracco in
```
Comment (AlexWaygood) [inline Doc/whatsnew/3.12.rst:783]: Given that this attribute is undocumented (and I'd prefer it to stay that way, though that's not a _strong_ opinion), could we add this something like "this ensures that `types.get_original_bases(typeddict_class)` will now always return the correct result"?

Comment (JelleZijlstra) [reply Doc/whatsnew/3.12.rst:783]: Maybe we could say "now work reliably with :func:`types.get_original_bases`", although that's a bit of a vacuous statement as `types.get_original_bases` didn't exist before 3.12 :)

Comment (AlexWaygood) [reply Doc/whatsnew/3.12.rst:783]: Eh, I guess you're right, it's probably fine as it is

**Review by AlexWaygood — APPROVED** [review]

**Review by stroxler — APPROVED** [review]
The PEP 698 section looks good to me, thanks!

**Review by hauntsaninja — APPROVED** [review]

File: Doc/whatsnew/3.12.rst [hunk 3]
```diff
@@ -270,7 +270,36 @@ typed dictionaries::
...
+module. It indicates to type checkers that the method is intended to override
+a method in a superclass. This allows type checkers to catch mistakes where
+a method that is intended to override something in a base class
+does not in fact do so.
+
+Example::
+
+   from typing import override
+
+   class Base:
```
Comment (rinarakaki) [inline Doc/whatsnew/3.12.rst:288]: 4 space tab would be more consistent.

**Pull request merged**